firmware/application/metadata_file.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2023 Kyle Reed
+ *
+ * This file is part of PortaPack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "metadata_file.hpp"
+
+#include "convert.hpp"
+#include "file_reader.hpp"
+#include "string_format.hpp"
+#include <string_view>
+
+namespace fs = std::filesystem;
+using namespace std::literals;
+
+const std::string_view center_freq_name = "center_frequency"sv;
+const std::string_view sample_rate_name = "sample_rate"sv;
+
+fs::path get_metadata_path(const fs::path& capture_path) {
+    auto temp = capture_path;
+    return temp.replace_extension(u".TXT");
+}
+
+Optional<File::Error> write_metadata_file(const fs::path& path, capture_metadata metadata) {
+    File f;
+    auto error = f.create(path);
+
+    if (error)
+        return error;
+
+    error = f.write_line(std::string{center_freq_name} + "=" +
+                         to_string_dec_uint(metadata.center_frequency));
+    if (error)
+        return error;
+
+    // TODO: Why does this divide by 8? Leaving as is for back-compat, but it's odd.
```

Comment:
Only thing that cames to mind is maybe it was intended that the sample rate was in bits and they need it in bytes.

Reply:
Or has it to do with decimation ? (like explained here: https://www.reddit.com/r/RTLSDR/comments/enrojq/comment/feceyxn/)